### PR TITLE
chore(portal): Misc sign in UX fixes

### LINKED ISSUE #3331: Portal usability improvements
Tracking issue for feedback conveyed by 1.0 users.

- [x] #3332
- [x] #3333
- [x] #3334
- [x] #3389
- [ ] "Go to sites to add additional Resources"
- [ ] "Are you sure you want to revoke all tokens? This will immediately sign the actor out of all clients." on the Gateways page

### LINKED ISSUE #3389: Sign in form fixes
@thomaseizinger was testing on Android and got stuck at sign in because it wasn't clear that the slug still needed to be entered.

- Root sign in form should validate slug is present
- Move the `https://app.firezone.dev` prefix outside the editable area, perhaps using this: https://flowbite.com/docs/forms/input-field/#input-group
- Fix layout for Firefox/small screens (see below)
- Show email the user entered during sign in flow to catch typos

Screenshot 2024-01-24 at 1 44 29 PM

### PR DESCRIPTION
**Pull request by jamilbk**
Fixes #3389 
Fixes #3331 

Will need #3433 to merge for the entra logo to show up, but this will fail gracefully if it lands before then.

Screenshot 2024-02-05 at 5 36 17 AM
Screenshot 2024-02-05 at 5 33 30 AM
Screenshot 2024-02-05 at 5 33 13 AM

### CODE REVIEW

File: elixir/apps/web/lib/web/live/sign_in.ex [hunk 1]
```diff
@@ -200,31 +200,46 @@ defmodule Web.SignIn do
...
+
+  defp provider_icon(%{adapter: :openid_connect} = assigns) do
+    ~H"""
+    <img src={~p"/images/openid-logo.svg"} alt="OpenID Connect Logo" class="w-5 h-5 mr-2" />
+    """
+  end
+
+  defp provider_icon(%{adapter: :microsoft_entra} = assigns) do
+    ~H"""
+    <img src={~p"/images/entra-logo.svg"} alt="Microsoft Entra Logo" class="w-5 h-5 mr-2" />
```
Comment (AndrewDryga) [inline elixir/apps/web/lib/web/live/sign_in.ex]: If we do this we also might want to show this logo elsewhere in the UI, eg. on the IdP list or creation.

File: elixir/apps/web/lib/web/live/sign_in.ex [hunk 2]
```diff
@@ -200,31 +200,46 @@ defmodule Web.SignIn do
...
-          hover:bg-neutral-100 hover:text-neutral-900
-    ]}>
-      Sign in with <%= @provider.name %>
-    </a>
+    <.button
+      navigate={~p"/#{@account}/sign_in/providers/#{@provider}/redirect?#{@params}"}
+      class="w-full"
+      style="info"
+    >
+      <%= provider_icon(@provider) %> Sign in with&nbsp;<strong><%= @provider.name %></strong>
```
Comment (AndrewDryga) [inline elixir/apps/web/lib/web/live/sign_in.ex]: I think `&nbsp;` is bad practice, instead you just want to set the container to no-wrap: https://tailwindcss.com/docs/text-wrap

**Review by AndrewDryga — APPROVED** [review]
One small thing and LGTM

File: elixir/apps/web/lib/web/live/sign_in.ex (hunk 1 above)
Comment (jamilbk) [reply elixir/apps/web/lib/web/live/sign_in.ex]: I was also thinking that. Added it as a core component and added it to the Identity Providers -> New form:

Screenshot 2024-02-05 at 9 45 38 AM

**Pull request merged**